target/linux/generic/hack-5.15/290-net-dsa-mv88e6xxx-allow-built-in-when-PTP-disabled.patch

```diff
@@ -0,0 +1,44 @@
+From 04cdbf10182b4cb9726c67ad7ae8d05d0d6bed95 Mon Sep 17 00:00:00 2001
+From: "Johnny S. Lee" <foss@jsl.io>
+Date: Thu, 15 Dec 2022 17:49:04 +0800
+Subject: [PATCH] net: dsa: mv88e6xxx: allow built-in when PTP disabled
+
+Objects of PTP hardware timestamping are not linked when the support is
+disabled, therefore MV88E6xxx should not be prevented to be built-in
+unless PTP support is enabled.
```

Comment:
AFAIU the issue isn't that you cannot built-in MV88E6xxx if PTP support is disabled, the issue is that you cannot built-in MV88E6xxx if PTP is built as a module, because then it depends on the PTP module - regardless if PTP support for MV88E6xxx is enabled or not.

Reply:
After a second reading I now understand what you wanted to say. I think a bit of clarification that you are talking about PTP support for the driver (`NET_DSA_MV88E6XXX_PTP`) and not PTP support in general (`PTP_1588_CLOCK`) would be good.

Reply:
Those are written in the commit message of this PR.
The patch's re-worded in the new version